### ISSUE TITLE
not working with nextjs

### ISSUE THREAD

**Issue by geduldig** [ReactTooltip/react-tooltip#1199]
I'm trying to get react-tooltip working with nextjs 14 in either a server or a client component. Is neither supprorted?

Jonas

**Comment by gabrieljablonski** [comment]
It is supposed to work normally on Next.js 14. Hard to say what's wrong without more info. Do you get any errors on the console? Have you been able to get the tooltip to work at all on a non-Next.js project?

Please check our [troubleshooting page](https://react-tooltip.com/docs/troubleshooting) for common issues.

If you could provide a sample repo so we can reproduce the issue, it would help us figure out what's wrong.

**Comment by geduldig** [comment]
Thank you for that info. I was able to get it working. I think I was looking at an example for another similar project. 

One more question… Does react-tooltip only work in client components, not server components?

**Comment by gabrieljablonski** [comment]
>Does react-tooltip only work in client components, not server components?

It's been a while since I've worked with Next.js, so I don't quite recall the details. If I remember correctly, the only thing about server components that affected the tooltip was the usage of `useEffect()`, but we added a workaround for that (see [here](https://github.com/ReactTooltip/react-tooltip/blob/master/src/utils/use-isomorphic-layout-effect.ts)).

I believe it should work without any issues, but let us know if you see anything weird.
